Lab1 Files

animalvgamer opened the pull request:
I have adjusted the Lab1 files to rst, and removed reference to the sequential part of the assignment.

In the circuit file, I have left both the sample XOR circuit and the sample sequential circuit, if perhaps a student might find that useful.

Everything else is effectively unchanged.

animalvgamer committed:
Lab1 files added, rst updated not to have Sequential part

animalvgamer committed:
Lab1 files added, rst updated not to have Sequential part

animalvgamer committed:
Sequential Input Circuit removed from Lab 1 Given.circ

animalvgamer committed:
Lab1.rst reference to sequential part of assignment removed

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+.. raw:: latex
```
Let's remove this. Also, could you add ":Author:" (or, more correctly, ":Editor:" line at the beginning?

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
+The following is an example of implementing the XOR function using
+microcode. Here is the truth table for XOR
+:math:`\left( \oplus \right)`.
```
None of the math renders on github :/. For this one, you can use unicode: ⊕

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
++---+---+----------------------------------------+-------+
+| X | Y | Address                                | Value |
++===+===+========================================+=======+
+| 0 | 0 | :math:`00 \left( \mathtt{0x0} \right)` | 0     |
```
This, I think you can use ``0x0`` for inline verbatim. Though, this might not work inside a table, which is OK if it doesn't work.

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
++-----+-----+-----+-----+-----+-----+-----+-----+------+------+------+
+| In7 | In6 | In5 | In4 | In3 | In2 | In1 | In0 | Out2 | Out1 | Out0 |
++=====+=====+=====+=====+=====+=====+=====+=====+======+======+======+
```
Can you add a blank column between the In and Out so it's more clear where the transition is? Also, I'm not sure if I asked this, but can you change some of these numbers so people can't turn in the exact same file from last quarter?

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+Don’t be intimidated by the number of inputs when doing the
+combinational circuit. You can use Logisim’s **Analyze Circuit** tool,
+in the Project drop-down menu, to have Logisim build the circuit for
+you. To learn how to use it, click on Help :math:`\rightarrow` User’s
```
I think this can be replaced with a unicode character, again.

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
+.. raw:: latex
+
+   \vspace{12pt}
```
remove

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
+.. raw:: latex
+
+   \vspace{12pt}
```
remove

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
+   \vspace{12pt}
+
+``python tester.py correct.txt your.txt``
```
If you want a block here use:

`.. code-block:: python`

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
+.. raw:: latex
+
+   \vspace{12pt}
```
remove

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+
+.. raw:: latex
+
+   \vspace{12pt}
```
remove

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,412 @@
+   names of the signals.
+
+#. Reset your circuit by pressing Ctrl + R, or by going to Simulate
+   :math:`\rightarrow` Reset Simulation.
```
Latex...

powerjg reviewed (commented):
Mostly looks good. There's some lingering latex that doesn't render on github that needs to be fixed, a minor update to the table, and a couple of other small things. Thanks for doing this so quickly!

animalvgamer committed:
Update lab1.rst

animalvgamer committed:
Update lab1.rst

animalvgamer committed:
Update lab1.rst

animalvgamer committed:
Added column separating inputs and outputs.

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,396 @@
+|                       |                       | In7-0.                |
++-----------------------+-----------------------+-----------------------+
+| ROMCombinationalOutpu | 2                     | The concatenation of  |
+| t                     |                       | Out2-0 from the ROM   |
```
T should be on the other line... that's annoying :/. Sorry.

animalvgamer replied:
No yea I noticed that later and it looks silly on the table, my b.

powerjg commented on lab1/lab1.rst:
```diff
@@ -0,0 +1,396 @@
+
+   -  12.5% for correct Part 2.
+
+   -  25% for correct Part 3.
```
I think this needs to be updated 50/50? Pick what you think is best.

animalvgamer replied:
Did 50/50 because it's the same logic in two forms, doesn't make sense to weigh them differently.

powerjg reviewed (commented):
Very close. Thanks for the quick updates!

animalvgamer commented:
I updated the dates and replaced math with unicode just now.

I did add the extra column to separate the inputs and outputs on the table, but before I got started on changing numbers, I wanted to ask how much I should change them by -- that is, should I change the input values as well as output? Should I keep the same number of items, etc?

animalvgamer committed:
Update lab1.rst

powerjg commented:
@jlperona How do you normally change the numbers for this assignment? Also, what's your opinion on the new scoring breakdown?

I like Bradley's suggestion for the assembly stuff, so I'm not going to include it in this lab. Instead, I'm going to post it on its own, tell the students it's "optional homework" and include some simpler assembly stuff on the quiz.

animalvgamer commented:
@powerjg about the assembly, in the case of optional, I would probably want to add/emphasize "highly-recommended", since it's basically prep-work for Lab 2.

powerjg commented:
Agreed. And even more, it's preparation for the quiz, which will look similar.

Commit:
Remove extra sequential piece

Left over from previous

jlperona commented:
@powerjg I'm okay with the current breakdown, given that there's not much else to the assignment. Since the other one is now optional I would like to add a smaller piece to this assignment, say, another smaller 4 input / 2 output truth table for them to implement first. That way, this gets them to really  understand it.

@animalvgamer Leave the first eight inputs the same, change their outputs. You can change the inputs for the rest of the individual ones and the longer ones with don't cares (though I would keep the total number of entries in the table roughly the same). Randomize the outputs for everything. After you do this, update the log files. It's probably easiest to implement the circuit in Logisim, then create the log file from within Logisim and add the X's back afterwards.

animalvgamer committed:
Update lab1.rst

Changed numbers, added 4 input-2 output ROM extra part,

animalvgamer commented:
So I just generated new numbers for the assignment, but I have yet to have a chance to do the correct log files. I need to get some rest for today but I will try to get those up tomorrow morning. I think that the pull request will be ready to merge once that's completed, if you guys are okay with the rst as it is now.

jlperona commented:
@animalvgamer Move the simpler one up so that it's the first problem chronologically. There should be four parts, two ROMs and two combinational, so you should have four log files at the end. I'd add headers to specify there's four parts that they need to do.

When you're generating, you only need to generate the ROM file and rename the signal name in the log file for the combinational one.

animalvgamer commented:
@jlperona I thought the goal of the simpler ROM assignment was just to get them comfortable with using a ROM for combinational logic, since we were assuming that they're coming from 154A with gate-based logic background. Thoughts?

jlperona commented:
@animalvgamer I see what you wanted now. The order is fine as is, and I'm okay with not doing the combinational logic for the simpler function. Make sure to add headers, even for part 3, since it slipped under the radar for me.

animalvgamer commented:
@jlperona What do you mean by headers?

animalvgamer committed:
Update lab1.rst

animalvgamer committed:
Update lab1.rst

powerjg commented:
Let me know when you think this is finalized and I'll look through it one last time before posting. I want this out tomorrow at noon *at the latest*.

Thanks for all of the effort so far!

animalvgamer committed:
Updated given circuit

animalvgamer committed:
Merge branch 'bwtemp' of https://github.com/jlpteaching/ECS154B into bwtemp

jlperona commented:
@animalvgamer The section titles would have been more appropriate to say. You say "part 1" at one point, but all the titles for each subpart has just a name, not "part X".

animalvgamer committed:
Numbered Assignment part titles

animalvgamer commented:
I still have to generate the new correct.txt files, but after that, I think that's everything.

jlperona commented:
Yup, I think the lab itself looks fine.

animalvgamer committed:
added updated correct.txt files

animalvgamer committed:
Merge branch 'bwtemp' of https://github.com/jlpteaching/ECS154B into bwtemp

animalvgamer commented:
Ok I think that's everything, I put a reference completed version of the circuit in the private repo, and the correct files are based on that. Feel free to review it as you see fit, I believe this should be ready to publish.

I'm heading our for tonight until late, so this may be the last update I can push for now.

powerjg commented:
LGTM. We may make minor changes, but I'll publish/announce this.

Thanks for all the hard work!!
